test/unit/slc/utest_slc.py

```diff
@@ -63,63 +70,84 @@ class SlcFactory:
+
+
+@pytest.fixture
+def slc_factory_create_or_open(tmp_path, git_repo_mock):
+    return SlcFactory(tmp_path, git_repo_mock, ScriptLanguageContainer.create_or_open)
+
+
+def _validate_slc(flavor_name, sample_slc_name, slc, path):
+    assert slc.language_alias == f"custom_slc_{sample_slc_name}"
+    assert slc.secrets.SLC_FLAVOR_CUDA == flavor_name
+    checkout_dir = path / constants.WORKSPACE_DIR / sample_slc_name / "git-clone"
+    assert slc.checkout_dir == checkout_dir
+    assert slc.flavor_path.is_dir()
+    assert (
+        slc.flavor_path
+        == checkout_dir / constants.FLAVORS_PATH_IN_SLC_REPO / flavor_name
+    )
+    assert slc.custom_pip_file.parts[-3:] == (
+        "flavor_customization",
+        "packages",
+        "python3_pip_packages",
+    )
+    assert slc.compression_strategy == CompressionStrategy.GZIP
+
+
+@pytest.fixture
+def slc_factory_create_select_method(request, tmp_path, git_repo_mock):
+    return SlcFactory(tmp_path, git_repo_mock, request.param)
 
 
+@pytest.mark.parametrize(
+    "slc_factory_create_select_method",
+    [ScriptLanguageContainer.create, ScriptLanguageContainer.create_or_open],
+    indirect=True,
+)
 def test_create(
     sample_slc_name,
-    monkeypatch: MonkeyPatch,
     caplog,
-    slc_factory,
+    slc_factory_create_select_method,
 ):
     flavor = "Strawberry"
-    with slc_factory.context(sample_slc_name, flavor) as testee:
+    with slc_factory_create_select_method.context(sample_slc_name, flavor) as testee:
         pass
 
-    assert slc_factory.git_repo_mock.clone_from.called
+    assert slc_factory_create_select_method.git_repo_mock.clone_from.called
     assert "Cloning into" in caplog.text
     assert "Fetching submodules" in caplog.text
 
-    assert testee.language_alias == f"custom_slc_{sample_slc_name}"
-    assert testee.secrets.SLC_FLAVOR_CUDA == flavor
-    checkout_dir = (
-        slc_factory.path / constants.WORKSPACE_DIR / sample_slc_name / "git-clone"
-    )
-    assert testee.checkout_dir == checkout_dir
-    assert testee.flavor_path.is_dir()
-    assert (
-        testee.flavor_path == checkout_dir / constants.FLAVORS_PATH_IN_SLC_REPO / flavor
-    )
-    assert testee.custom_pip_file.parts[-3:] == (
-        "flavor_customization",
-        "packages",
-        "python3_pip_packages",
+    _validate_slc(
+        flavor, sample_slc_name, testee, slc_factory_create_select_method.path
     )
-    assert testee.compression_strategy == CompressionStrategy.GZIP
 
 
-def test_repo_missing(sample_slc_name):
-    secrets = SlcSecretsMock.create(sample_slc_name, Path())
-    with pytest.raises(SlcError, match="SLC Git repository not checked out"):
-        ScriptLanguageContainer(secrets, sample_slc_name)
+def test_repo_missing(sample_slc_name, tmp_path):
+    with current_directory(tmp_path):
+        secrets = SlcSecretsMock.create(sample_slc_name, "aflavor")
+        with pytest.raises(SlcError, match="SLC Git repository not checked out"):
```

Comment:
If I understand this correctly, you don't fix the clone issue in this PR

Reply:
Exactly, the clone fix will come in a later PR